packages/ai-jsx/src/batteries/constrained-output.tsx

```diff
@@ -336,9 +336,16 @@ export async function* JsonChatCompletionFunctionCall(
     </ChatCompletion>
   );
 
-  const frames = render(childrenWithCompletion);
+  const frames = render(childrenWithCompletion, { stop: (e) => e.tag === FunctionCall, map: (e) => e });
   for await (const frame of frames) {
-    const object = JSON.parse(frame).arguments;
+    const functionCall = frame.find((e) => AI.isElement(e) && e.tag === FunctionCall) as
+      | AI.Element<AI.PropsOfComponent<typeof FunctionCall>>
+      | undefined;
+    if (!functionCall) {
+      continue;
+    }
+
+    const object = functionCall.props.args;
```

Comment:
I forgot to apply this comment from Nick on previous PR, can you rename `object` to `jsonResult` too please?
```suggestion
    const jsonResult = functionCall.props.args;
```